plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeSystemTables.java

```diff
@@ -103,4 +103,83 @@ public void testPropertiesTable()
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableMultipleColumns()
+    {
+        String tableName = "test_partitions_table_multiple_columns";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE, _varchar VARCHAR) WITH (partitioned_by = ARRAY['_date', '_varchar'] )");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE), 'a'), (1, CAST('2019-09-09' AS DATE), 'b'), (2, CAST('2019-09-09' AS DATE), 'c')", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE), 'd'), (4, CAST('2019-09-10' AS DATE), 'e'), (5, CAST('2019-09-10' AS DATE), 'f'), (4, CAST('2019-09-10' AS DATE), 'f')", 4);
+            assertUpdate("UPDATE " + tableName + " SET _bigint = 50 WHERE _bigint =  BIGINT '5'", 1);
+            assertUpdate("DELETE FROM " + tableName + " WHERE _date =  DATE '2019-09-08'", 1);
+            assertQuerySucceeds("ALTER TABLE " + tableName + " EXECUTE OPTIMIZE");
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 6");
+            assertQuery("SELECT count(*) FROM \"" + tableName + "$partitions\"", "VALUES 5");
+            assertQuery("SELECT count(partition._varchar) FROM \"" + tableName + "$partitions\"", "VALUES 5");
+            assertQuery("SELECT count(distinct partition._date) FROM \"" + tableName + "$partitions\"", "VALUES 2");
+
+            assertQuery("SHOW COLUMNS FROM \"" + tableName + "$partitions\"",
+                    """
+                            VALUES
+                            ('partition', 'row(_date date, _varchar varchar)', '', ''),
+                            ('file_count', 'bigint', '', ''),
+                            ('total_size', 'bigint', '', '')
+                            """);
+
+            assertQuery(
+                "SELECT partition._date, partition._varchar FROM \"" + tableName + 
+                "$partitions\"", "VALUES (DATE '2019-09-09', 'b'), (DATE '2019-09-09', 'c'), (DATE '2019-09-09', 'd'), (DATE '2019-09-10', 'e'), (DATE '2019-09-10', 'f')");
+        }
+        finally{
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        }
+    }
+
+    @Test
+    public void testPartitionsTableUnPartitioned()
+    {
+        String tableName = "test_partitions_table_unpartitioned";
+        try {
+            assertUpdate("CREATE TABLE " + tableName + " (_bigint BIGINT, _date DATE)");
+            assertUpdate("INSERT INTO " + tableName + " VALUES (0, CAST('2019-09-08' AS DATE)), (1, CAST('2019-09-09' AS DATE)), (2, CAST('2019-09-09' AS DATE))", 3);
+            assertUpdate("INSERT INTO " + tableName + " VALUES (3, CAST('2019-09-09' AS DATE)), (4, CAST('2019-09-10' AS DATE)), (5, CAST('2019-09-10' AS DATE))", 3);
+            assertQuery("SELECT count(*) FROM " + tableName, "VALUES 6");
```

Comment:
Use rather `SELECT *` instead of `SELECT count(*)`  - you are hiding this way the fact that the relation has no columns.

Reply:
Added additional test to address this case: `assertQueryReturnsEmptyResult("SELECT * FROM \"" + tableName + "$partitions\"");`

Reply:
Resolved